queries.sql

```diff
@@ -23,3 +23,195 @@ WHERE name <> 'Gabumon';
+
+/* Who owns the most animals? */
+SELECT full_name FROM owners
+JOIN animals ON owners.id = owner_id
+GROUP BY full_name
+ORDER BY COUNT(*)
+DESC LIMIT 1;
+
+/* DAY 4 */
+/* Write queries to answer the following */
+
+/* Who was the last animal seen by William Tatcher? */
+SELECT vets.name AS "Vet Name", animals.name AS "Animal Name", visits.date_of_visit
+  FROM vets
+  INNER JOIN visits
+    ON vets.id = visits.vet_id
+  INNER JOIN animals
+    ON visits.animal_id = animals.id
+  WHERE vets.name = 'William Tatcher'
+  ORDER by date_of_visit DESC LIMIT 1;
+
+/* How many different animals did Stephanie Mendez see? */
+SELECT vets.name AS "Vet Name", COUNT(DISTINCT animals.name)
+   FROM vets
+   INNER JOIN visits
+     ON vets.id = visits.vet_id
+   INNER JOIN animals
+     ON visits.animal_id = animals.id
+   WHERE vets.name = 'Stephanie Mendez'
+   GROUP BY vets.name;
+
+
+/* List all vets and their specialties, including vets with no specialties. */
+SELECT vets.name AS "Vet Name", species.name AS "Species"
+  FROM vets
+  FULL OUTER JOIN specializations
+    ON vets.id = specializations.vet_id
+  FULL OUTER JOIN species
+    ON species.id = specializations.species_id;
+
+/* List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. */
+SELECT animals.name AS "Animal", visits.date_of_visit
+  FROM animals
+  JOIN visits
+    ON animals.id = visits.animal_id
+  WHERE visits.vet_id = (SELECT id FROM vets WHERE vets.name = 'Stephanie Mendez')
+  AND
+  visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+/* What animal has the most visits to vets? */
+SELECT animals.name, COUNT(*)
+  FROM animals
+  JOIN visits
+    ON animals.id = visits.animal_id
+  GROUP BY animals.name
+  ORDER BY count DESC LIMIT 1;
+
+/*Who was Maisy Smith's first visit?*/
+SELECT vets.name AS "Vet", animals.name AS "Animal", visits.date_of_visit
+  FROM vets
+  JOIN visits
+    ON vets.id = visits.vet_id
+  JOIN animals
+    ON visits.animal_id = animals.id
+  WHERE vets.name = 'Maisy Smith'
+  ORDER BY date_of_visit LIMIT 1;
+
+/* Details for most recent visit: animal information, vet information, and date of visit. */
+SELECT
+  vets.id AS "Vet id", vets.name AS "Vet", date_of_graduation,
+  visits.date_of_visit,
+  animals.id AS "Animal id", animals.name AS "Animal", date_of_birth, escape_attempts, neutered, weight_kg, species_id, owner_id
+  FROM vets
+  JOIN visits
+    ON vets.id = visits.vet_id
+  JOIN animals
+    ON visits.animal_id = animals.id
+  ORDER BY date_of_visit DESC LIMIT 1;
+
+/* How many visits were with a vet that did not specialize in that animal's species? */
+SELECT vets.name AS "Vet", COUNT(*) FROM visits 
+LEFT JOIN vets ON visits.vet_id = vets.id
+LEFT JOIN specializations ON vets.id = specializations.vet_id 
+LEFT JOIN species ON specializations.species_id = species.id 
+WHERE specializations.species_id IS NULL OR specializations.species_id != species.id 
+GROUP BY vets.name;
+
```

Comment:
- [ ] Kindly add the snapshot of this query in your PR to reflect the chnages as suggested by the last reviwer 👍